File: app/src/main/java/mrkm4ntr/twitterclient/sync/TwitterSyncService.java
package mrkm4ntr.twitterclient.sync;

import android.app.Service;
import android.content.Intent;
import android.os.IBinder;
import android.support.annotation.Nullable;
import android.util.Log;

public class TwitterSyncService extends Service {

    private static final Object SYNC_ADAPTER_LOCK = new Object();
    private static TwitterSyncAdapter sTwitterSyncAdapter = null;

    @Override
    public void onCreate() {
        Log.d("TwitterSyncService", "onCreate - TwitterSyncService");
        synchronized (SYNC_ADAPTER_LOCK) {
            if (sTwitterSyncAdapter == null) {
                sTwitterSyncAdapter = new TwitterSyncAdapter(getApplicationContext(), true);
            }
        }
        super.onCreate();
    }

    @Nullable
    @Override
    public IBinder onBind(Intent intent) {
        return sTwitterSyncAdapter.getSyncAdapterBinder();
    }
}

File: app/src/main/kotlin/mrkm4ntr/twitterclient/sync/TwitterSyncService.kt
package mrkm4ntr.twitterclient.sync

import android.app.Service
import android.content.Intent
import android.os.IBinder
import android.util.Log

class TwitterSyncService : Service() {

    override fun onCreate() {
        Log.d("TwitterSyncService", "onCreate - TwitterSyncService")
        synchronized (SYNC_ADAPTER_LOCK) {
            if (sTwitterSyncAdapter == null) {
                sTwitterSyncAdapter = TwitterSyncAdapter(applicationContext, true)
            }
        }
        super.onCreate()
    }

    override fun onBind(intent: Intent): IBinder? {
        return sTwitterSyncAdapter!!.syncAdapterBinder
    }

    companion object {

        private val SYNC_ADAPTER_LOCK = Object()
        private var sTwitterSyncAdapter: TwitterSyncAdapter? = null
    }
}
